go/database/mpt/rlp/rlp.go

```diff
@@ -50,6 +51,90 @@ func EncodeInto(dst []byte, item Item) []byte {
 	return item.write(writer)
 }
 
+// Decode decodes an RLP stream into an item.
+func Decode(rlp []byte) (Item, error) {
+	item, consumed, err := decode(rlp)
+	if consumed < uint64(len(rlp)) {
+		return nil, fmt.Errorf("unexpected trailing data: %x", rlp[consumed:])
+	}
+	return item, err
+}
+
+// decode decodes an RLP stream into an item.
+// It checks first byte of the RLP stream to determine the type of the item.
+// Based on the type, it decodes the type.
```

Comment:
nit: implementation details should go into the function body

Reply:
right, moved in the body